sql/src/main/java/io/crate/analyze/symbol/Literal.java

```diff
@@ -139,9 +139,11 @@ public boolean equals(Object obj) {
         if (obj == null || getClass() != obj.getClass()) return false;
         Literal literal = (Literal) obj;
         if (valueType().equals(literal.valueType())) {
-            if (valueType().compareValueTo(value, literal.value) == 0) {
```

Comment:
Well, then they're bugged too. Or we need to document that arguments to compareValueTo must not be null.